src/tests/ftest/util/telemetry_utils.py

```diff
@@ -1217,13 +1248,21 @@ def _get_data(self, names, info):
                 if name not in data:
                     data[name] = {}
                 for metric in metrics[name]['metrics']:
-                    if 'labels' not in metric or 'value' not in metric:
+                    if 'labels' not in metric or \
+                            ('value' not in metric and 'buckets' not in metric):
                         continue
                     labels = [f'host:{host}']
                     for key, value in metric['labels'].items():
                         labels.append(":".join([str(key), str(value)]))
                     label_key = ",".join(sorted(labels))
-                    data[name][label_key] = metric['value']
+                    if 'value' in metric:
+                        data[name][label_key] = metric['value']
+                    else:
+                        data[name][label_key] = {
+                            'sample_count': metric['sample_count'],
+                            'sample_sum': metric['sample_sum'],
+                            'buckets': metric['buckets'],
+                        }
         return data
```

Comment:
Maybe we should handle it this way:
```suggestion
                for metric in metrics[name]['metrics']:
                    if 'labels' not in metric:
                        continue
                    labels = [f'host:{host}']
                    for key, value in metric['labels'].items():
                        labels.append(":".join([str(key), str(value)]))
                    label_key = ",".join(sorted(labels))
                    if 'value' in metric:
                        data[name][label_key] = metric['value']
                    elif 'sample_count' in metric and 'sample_sum' in metric and 'buckets' in metric:
                        data[name][label_key] = {
                            'sample_count': metric['sample_count'],
                            'sample_sum': metric['sample_sum'],
                            'buckets': metric['buckets'],
                        }
                    else:
                        continue
        return data
```
Or will there always be an accompanying `value` entry for each `buckets` entry?

Reply:
I disagree with the `else..continue` because that could silently mask errors. I think keeping the implicit assumption is better because it will raise an exception if something is unexpected

Reply:
> I think keeping the implicit assumption is better because it will raise an exception if something is unexpected

Yeah, that's where I was going with this... The default/standard metric is the `value` type, and I've added a new `histogram` type to allow tests to have access to the histograms that are in the DFS metrics (and will probably be added elsewhere now that they're supported properly).

Reply:
I see what you mean now. Should we handle this as part of the other ftest updates?